crates/cxx-qt-lib/src/gui/qimage.rs

```diff
@@ -11,12 +11,65 @@ mod ffi {
     unsafe extern "C++" {
         include!("cxx-qt-lib/qimage.h");
         type QImage = super::QImage;
+        include!("cxx-qt-lib/qsize.h");
+        type QSize = crate::QSize;
+        include!("cxx-qt-lib/qrect.h");
+        type QRect = crate::QRect;
+        include!("cxx-qt-lib/qcolor.h");
+        type QColor = crate::QColor;
 
         /// Whether the QImage is null.
         ///
         /// This means that the QImage has all parameters set to zero and no allocated data.
         #[rust_name = "is_null"]
         fn isNull(self: &QImage) -> bool;
+
+        /// Returns true if all the colors in the image are shades of gray
+        #[rust_name = "all_gray"]
+        fn allGray(self: &QImage) -> bool;
+
+        /// For 32-bit images, this function is equivalent to allGray().
+        /// For color indexed images, this function returns true if color(i) is QRgb(i, i, i)
+        /// for all indexes of the color table; otherwise returns false.
+        #[rust_name = "is_gray_scale"]
+        fn isGrayscale(self: &QImage) -> bool;
+
+        /// Returns true if the image has a format that respects the alpha channel, otherwise returns false.
+        #[rust_name = "has_alpha_channel"]
+        fn hasAlphaChannel(self: &QImage) -> bool;
+
+        /// Returns the size of the color table for the image.
+        #[rust_name = "color_count"]
+        fn colorCount(self: &QImage) -> i32;
+
+        /// Returns the depth of the image.
+        fn depth(self: &QImage) -> i32;
+
+        /// Returns the height of the image.
+        fn height(self: &QImage) -> i32;
+
+        /// Returns the width of the image.
+        fn width(self: &QImage) -> i32;
+
+        /// Returns the size of the image.
+        fn size(self: &QImage) -> QSize;
+
+        /// Returns the enclosing rectangle (0, 0, width(), height()) of the image.
+        fn rect(self: &QImage) -> QRect;
+
+        /// Returns the color of the pixel at coordinates (x, y) as a QColor.
+        #[rust_name = "pixel_color"]
+        fn pixelColor(self: &QImage, x: i32, y: i32) -> QColor;
+
+        /// Returns the number of pixels that fit horizontally in a physical meter.
+        #[rust_name = "dots_per_meterx"]
+        fn dotsPerMeterX(self: &QImage) -> i32;
+
+        /// Returns the number of pixels that fit vertically in a physical meter.
+        #[rust_name = "dots_per_metery"]
+        fn dotsPerMeterY(self: &QImage) -> i32;
+
+        fn valid(self: &QImage, x: i32, y: i32) -> bool;
```

Comment:
don't forget doc lines :-) 

there is also the fun of Rust not supporting overloads, so for this method we could support QPoint or x,y. Either we support both with a suffix on one of them (eg `_with_point`) or support one of them and just use `point.x, point.y`. Depends how often we think this method would be used. `x, y` is probably enough.

Reply:
yep I saw that we can't provide overloads methods :( 
For the moment x,y is enough I think.

Reply:
Another way to go about this is to template the method:
```rust
fn valid(&self, point: Into<QPoint>) -> bool { ... }
```
That way you get sort-off overloading in Rust.
Assume `(i32, i32)` supported `Into<QPoint>`.

Then you could do either:
```
qimage.valid(my_qpoint);
qimage.valid((x, y)); // <-- Note the double-parens to create a tuple.
```

Though I guess if you didn't template the method it could be:
```
qimage.valid((x, y).into())
```
Which isn't terrible either.

I guess the takeaway here is that QPoint should implement `From<(i32, i32)>`